internal/controller/resources/redis.go

```diff
@@ -33,13 +39,15 @@ func (in *RedisInstanceMetadataProvider) GetResourceLocation(r *redisv1beta1.Red
 	return r.Spec.Region
 }
 
-func (in *RedisInstanceMetadataProvider) GetResourceID(projectID string, r *redisv1beta1.RedisInstance) string {
+func (in *RedisInstanceMetadataProvider) GetResourceID(projectInfo *resourcemanagerpb.Project, r *redisv1beta1.RedisInstance) string {
 	name := r.Name
 	if r.Spec.ResourceID != nil {
 		name = *r.Spec.ResourceID
 	}
 
 	region := r.Spec.Region
 
-	return fmt.Sprintf("//redis.googleapis.com/projects/%s/locations/%s/instances/%s", projectID, region, name)
+	projectNumber := projectNumRe.FindString(projectInfo.Name)
```

Comment:
```suggestion
	projectNumber := strings.TrimPrefix(projectInfo.Name, "projects/")
```

Since the API clearly defines that this will always be the numeric ID prefixed by `projects/`, I would lean towards a simpler solution here. Makes it easier to understand and also gives better performance than a regex.